crates/composefs/src/fsverity/mod.rs

```diff
@@ -45,6 +55,33 @@ pub enum CompareVerityError {
     DigestMismatch { expected: String, found: String },
 }
 
+/// An owned file descriptor with fsverity enabled.  Used in contexts
+/// where a user-supplied file descriptor may be returned back to the
+/// user directly (`Orig`), or a distinct copy of the file descriptor
+/// (`Copy`) may be returned in place of the original.
+#[derive(Debug)]
+pub enum VerityFd {
+    Orig(OwnedFd),
+    Copy(OwnedFd),
+}
+
+impl VerityFd {
+    pub fn is_orig(&self) -> bool {
+        matches!(self, Self::Orig(_))
+    }
+
+    pub fn is_copy(&self) -> bool {
```

Comment:
I think we should have docstrings for public APIs. But it's not clear to me these need to be public at all given it's just used in the tests.